src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -200,10 +235,16 @@ public TopicPartitionChannel(
     this.channel = Preconditions.checkNotNull(openChannelForTable());
     this.kafkaRecordErrorReporter = Preconditions.checkNotNull(kafkaRecordErrorReporter);
     this.sinkTaskContext = Preconditions.checkNotNull(sinkTaskContext);
+    this.conn = conn;
+    if (conn != null) {
```

Comment:
when the conn could be null?

Reply:
When the topicPartitionChannel is used in tests